cmd/snap/cmd_snapshot.go

```diff
@@ -384,4 +391,67 @@ func init() {
 				desc: i18n.G("The snap for which data will be verified"),
 			},
 		})
+
+	cmd := addCommand("export-snapshot",
+		shortExportSnapshotHelp,
+		longExportSnapshotHelp,
+		func() flags.Commander {
+			return &exportSnapshotCmd{}
+		}, nil, []argDesc{
+			{
+				name: "<id>",
+				// TRANSLATORS: This should not start with a lowercase letter.
+				desc: i18n.G("Set id of snapshot to export"),
+			},
+			{
+				name: "<filename>",
+				// TRANSLATORS: This should not start with a lowercase letter.
+				desc: i18n.G("The filename of the export"),
+			},
+		})
+	cmd.hidden = true
+}
+
+type exportSnapshotCmd struct {
+	clientMixin
+	Positional struct {
+		ID       snapshotID `positional-arg-name:"<id>"`
+		Filename string     `long:"filename"`
+	} `positional-args:"yes" required:"yes"`
+}
+
+func (x *exportSnapshotCmd) Execute([]string) error {
+	setID, err := x.Positional.ID.ToUint()
+	if err != nil {
+		return err
+	}
+
+	r, expectedSize, err := x.client.SnapshotExport(setID)
+	if err != nil {
+		return err
+	}
+
+	filename := x.Positional.Filename
+	f, err := os.Create(filename + ".part")
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	defer func() { os.Remove(filename + ".part") }()
```

Comment:
I think that's right but should not make a difference in practise.  If the .part file is there it should be removed. If things worked it will have been renamed to the real filename.